openml/tasks/task.py

```diff
@@ -36,32 +38,28 @@ def get_dataset(self):
     def get_X_and_y(self):
         dataset = self.get_dataset()
         # Replace with retrieve from cache
-        if 'Supervised Classification'.lower() in self.task_type.lower():
+        if self.task_type_id == 1:
+        # if 'Supervised Classification'.lower() in self.task_type.lower():
             target_dtype = int
-        elif 'Supervised Regression'.lower() in self.task_type.lower():
+        # elif 'Supervised Regression'.lower() in self.task_type.lower():
+        elif self.task_type_id == 2:
             target_dtype = float
+        # elif ''.lower('Learning Curve') in self.task_type.lower():
+        elif self.task_type_id == 3:
+            target_dtype = int
```

Comment:
Maybe you want to make [the specification](https://www.openml.org/tt/3) more clear.